tests/keras/test_callbacks.py

```diff
@@ -909,5 +909,84 @@ def tests_RemoteMonitorWithJsonPayload():
                   validation_data=(X_test, y_test), callbacks=cbks, epochs=1)
 
 
+def mxnet_model_checkpoint_test_helper(monitor, save_best_only, mode, prefix='test', epochs=1):
+    np.random.seed(1337)
+    (X_train, y_train), (X_test, y_test) = get_test_data(num_train=train_samples,
+                                                         num_test=test_samples,
+                                                         input_shape=(input_dim,),
+                                                         classification=True,
+                                                         num_classes=num_classes)
+    y_test = np_utils.to_categorical(y_test)
+    y_train = np_utils.to_categorical(y_train)
+    model = Sequential()
+    model.add(Dense(num_hidden, input_dim=input_dim, activation='relu'))
+    model.add(Dense(num_classes, activation='softmax'))
+    model.compile(loss='categorical_crossentropy',
+                  optimizer='rmsprop',
+                  metrics=['accuracy'])
+
+    cbks = [callbacks.MXNetModelCheckpoint(prefix, monitor=monitor,
+                                           save_best_only=save_best_only, mode=mode)]
+    model.fit(X_train, y_train, batch_size=batch_size,
+              validation_data=(X_test, y_test), callbacks=cbks, epochs=epochs)
+
+
+@pytest.mark.skipif((K.backend() != 'mxnet'),
+                    reason='Supported for MXNet backend only.')
+@keras_test
+def test_mxnet_model_checkpoint_save_all_auto_mode():
+    mxnet_model_checkpoint_test_helper(monitor='val_loss', save_best_only=False, mode='auto')
+    assert os.path.isfile('test-symbol.json')
```

Comment:
Not sure how other files are handled in unit tests - but are we not verifying if the content is as expected?

Reply:
Yes as described earlier, here we are generating random data and verifying the flow of callbacks and confirming if all required artifacts are generated. This depends on save_mxnet_model API for the content of file. save_mxnet_model API tests are covering the content is as expected.